employee-rostering-backend/src/main/java/org/optaweb/employeerostering/service/roster/RosterService.java

```diff
@@ -50,12 +116,294 @@ public RosterState getRosterState(Integer tenantId) {
+    public ShiftRosterView getShiftRosterView(final Integer tenantId, Integer pageNumber, Integer numberOfItemsPerPage,
+                                              final String startDateString,
+                                              final String endDateString) {
+
+        return getShiftRosterView(tenantId, LocalDate.parse(startDateString), LocalDate.parse(endDateString),
+                                  Pagination.of(pageNumber, numberOfItemsPerPage));
+    }
+
+    private ShiftRosterView getShiftRosterView(final Integer tenantId,
+                                               final LocalDate startDate,
+                                               final LocalDate endDate,
+                                               final Pagination pagination) {
+
+        Pageable spotPage = PageRequest.of(pagination.getPageNumber(), pagination.getNumberOfItemsPerPage());
+        final List<Spot> spots = spotRepository.findAllByTenantId(tenantId, spotPage);
+
+        return getShiftRosterView(tenantId, startDate, endDate, spots);
+    }
+
+    @Transactional
+    public ShiftRosterView getShiftRosterViewFor(Integer tenantId, String startDateString, String endDateString,
+                                                 List<Spot> spots) {
+        LocalDate startDate = LocalDate.parse(startDateString);
+        LocalDate endDate = LocalDate.parse(endDateString);
+        if (null == spots) {
+            throw new IllegalArgumentException("spots is null!");
+        }
+
+        return getShiftRosterView(tenantId, startDate, endDate, spots);
+    }
+
+    private ShiftRosterView getShiftRosterView(Integer tenantId, LocalDate startDate, LocalDate endDate,
+                                               List<Spot> spotList) {
+        ShiftRosterView shiftRosterView = new ShiftRosterView(tenantId, startDate, endDate);
+        shiftRosterView.setSpotList(spotList);
+        List<Employee> employeeList = employeeRepository.findAllByTenantId(tenantId, PageRequest.of(0,
+                                                                                                    Integer.MAX_VALUE));
+        shiftRosterView.setEmployeeList(employeeList);
+
+        Set<Spot> spotSet = new HashSet<>(spotList);
+        ZoneId timeZone = getRosterState(tenantId).getTimeZone();
+
+        List<Shift> shiftList = shiftRepository.filterWithSpots(tenantId, spotSet,
+                                                                startDate.atStartOfDay(timeZone).toOffsetDateTime(),
+                                                                endDate.atStartOfDay(timeZone).toOffsetDateTime());
+
+        Map<Long, List<ShiftView>> spotIdToShiftViewListMap = new LinkedHashMap<>(spotList.size());
+        // TODO FIXME race condition solverManager's bestSolution might differ from the one we just fetched, so the
+        //  score might be inaccurate
+        Roster roster = solverManager.getRoster(tenantId);
+        if (roster == null) {
```

Comment:
Inconsistent use of Yoda; you used a Yoda condition above, but this condition is not Yoda.